java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/EmailSenderService.java

```diff
@@ -1,285 +1,103 @@
-
-				// smtp.setAuthentication("username", "password"); // optional SMTP auth
-
-				String name = mail.getFrom().getName();
-				String from = mail.getFrom().getEmail();
-				String subject = mail.getSubject();
-				String body = mail.getContent().get(0).getValue(); // html content
-				String to = getTo();
-
-				smtp.setFrom(from, name);
-				smtp.setSubject(subject);
-				smtp.setMsg(body);
-				smtp.addTo(to);
-				smtp.send();
-
-			} catch(EmailException e) {
-				LOGGER.warn("SMTP error: " + e.getMessage() + " (" + e.getCause().getMessage() + ")");
-				return fakeSuccess();
-			}
-
-			return new Response(SENDGRID_SUCCESS_STATUS_CODE, "", null);
-		}
+	public static void test(String[] args) throws FileNotFoundException, UnsupportedEncodingException {
+		EmailSenderService sender = new EmailSenderService();
+		String email = "dmarc-reports@osmand.net";
+//		sender.sendOsmRecipientsDeleteEmail(email); // obsolete
+//		sender.sendOsmAndCloudPromoEmail(email, "promo"); // test ok
+//		sender.sendOsmAndCloudWebEmail(email, "token", "action"); // test ok
+//		sender.sendOsmAndCloudRegistrationEmail(email, "token", true); // test ok
+//		sender.sendPromocodesEmails(email, "promocode/android", "ANDROID"); // test ok
+//		sender.sendPromocodesEmails(email, "promocode/ios", "IOS"); // test ok
 	}
 
-	private EmailSenderService() {
-		final String smtpServer = System.getenv("SMTP_SERVER");
-		final String apiKey = System.getenv("SENDGRID_KEY");
-		wrapper = new SmtpSendGridSender(smtpServer, apiKey);
-	}
-
-    public void sendOsmAndCloudPromoEmail(String email, String promo) {
-		LOGGER.info("Sending mail to: " + email);
-		Email from = new Email(NOREPLY_MAIL_FROM);
-		from.setName("OsmAnd");
-		Email to = new Email(email);
-		String topic = "Congratulations and welcome to OsmAnd Cloud!";
-		StringBuilder contentStr = new StringBuilder();
-		contentStr.append("Dear OsmAnd User!");
-		contentStr.append("<br><br>");
-		contentStr.append("You have been selected for OsmAnd Cloud promo subscription. Your promo OsmAnd Pro is <b>"+promo+"</b><br>");
-		contentStr.append("Now you can open OsmAnd Settings -> Backup and Restore and Login with this email to OsmAnd Cloud and get all features enabled.<br>");
-		contentStr.append("<br><br>");
-		contentStr.append("Best Regards, <br>OsmAnd Team");
+    private static final Log LOGGER = LogFactory.getLog(EmailSenderService.class);
 
-		Content content = new Content("text/html", contentStr.toString());
-		Mail mail = new Mail(from, topic, to, content);
-		mail.from = from;
-		try {
-			Response response = wrapper.send(mail);
-			LOGGER.info("Response code: " + response.getStatusCode());
-		} catch (Exception e) {
-			LOGGER.warn(e.getMessage(), e);
-		}
+    public void sendOsmAndCloudPromoEmail(String email, String promo) throws FileNotFoundException, UnsupportedEncodingException {
+		boolean ok = new EmailSenderTemplate()
+				.load("cloud/promo")
```

Comment:
No need lang here?

Reply:
lang is unknown for cloud/promo templates, so not used